servers/src/test/java/tachyon/master/journal/JsonJournalFormatterTest.java

```diff
@@ -15,9 +15,119 @@
 
 package tachyon.master.journal;
 
+import java.io.File;
+import java.io.IOException;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
 public class JsonJournalFormatterTest extends JournalFormatterTestBase {
+
+  public static final String JSON_SAMPLE_PATH = "/JournalEntries.json";
+
+  private static JsonNode sRootNode;
+
   @Override
   protected JournalFormatter getFormatter() {
     return new JsonJournalFormatter();
   }
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    String entriesFile =
+        JsonJournalFormatterTest.class.getResource(JSON_SAMPLE_PATH).getFile();
+    sRootNode = new ObjectMapper().readTree(new File(entriesFile));
+  }
+
+  // given an entry type, it checks if a given json is deserialized properly
+  private final void entryJsonTest(JournalEntryType type) throws IOException {
+    JsonNode n = sRootNode.get(type.toString());
+    new ObjectMapper().writeValue(mOs, n);
+    JournalEntry readEntry = read();
+    assertSameEntry(mDataSet.get(type), readEntry);
+  }
+
+  // Block
+
+  @Test
+  public void blockIdGeneratorEntryJsonTest() throws IOException {
```

Comment:
Other than enumerating all the values one function per type, what about iterating over the key set of JournalFormatterTestBase's dataset?